specs/tree-view/tree-view.md

```diff
@@ -0,0 +1,195 @@
+
+### Background
+
+We recently created a React implementation of Tree View and Tree View Item with [PR #2510](https://github.com/microsoft/fast-dna/pull/2510). 
+
+Currently, a react implementation exists which supports a `defaultExpanded` value which sets the default expanded or collapsed state. In the event that the `defaultExpanded` value is false, but there is a selected child the tree will expand automatically. [Issue #2597](https://github.com/microsoft/fast-dna/issues/2597) requests an expanded prop to fully control the expanded/collapsed state.
+
+### Use Cases
+
+Primarily used to present a hierarchical list such as a file system. A tree view can also be served as a flat list where the items will function as navigation.
+  
+### Features
+
+- **Selection:** A tree view can support either single select or multi-select behavior for nodes. **I would propose that for the initial spec we start with supporting single select only.*
+
+
+- **Drag & Drop support:** Often, as is in the case of a "file system" example, drag and drop support is desired. This does present [challenges](#drag-&-drop).
+
+### Risks and Challenges
+
+#### Nesting:
+
+Infinite nesting combined with the current visual design presents a styling challenge. Consider a scenario where we have the following tree in pseudo code:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>
+        Parent Node && Root node
+        <fast-tree-item>Nested node one</fast-tree-item>
+        <fast-tree-item>Nested node two</fast-tree-item>
+    </fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+</fast-tree-view>
+```
+In this scenario, there is no root node with an expand/collapse glyph. The third item in the tree view is a parent node, which *would* have an expand/collapse glyph to the right. To ensure that the content of the parent node align with the rest of the root nodes, the entire tree will need to visually shift to ensure alignmnet of the tree content. This presents an interesting challenge for how we will communicate these changes to the parent to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has child nodes.
+
+#### Drag & Drop
+
+Drag & Drop support poses a unique set of challenges, especially if our goal is to provide out of the box support via the [HTML Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API). Alternatively, those using our components with React, Vue, Angular, etc may want to use libraries built to work with their specific framework flavor. 
+
+**Considering the challenge that drag & drop would present for any given component, I think it's worth holding on that feature, and taking it up seperately.*
```

Comment:
I think it would be better to just file an issue for drag and drop and link it here.

Reply:
I thought about that, but I think it's helpful to include here for a couple of reasons. Drag and drop is a feature that we've been asked for previously as well as improvements to "other" (react) implementations. While it may not be initially included, I think it's important to list the feature as part of the spec to show our thought process and reasoning why it may not be included initially. 

I agree with filing an issue, but it may make more sense to do that when we create a PR for this control to close the issue. My primary reasoning is 1) we may end up building in support for D&D once we get further into development of this, and 2) the control will give context for that issue. I think we could create now, but there wouldn't be a ton of value to the issue outside a placeholder IMO.

Reply:
I'm adding a section in the template that might help with where this lives see: https://github.com/microsoft/fast-dna/pull/2712

My concern is that other components may need drag and drop and we should have an issue, even if it's mostly blank and marked as an investigation, that multiple specs can link back to. I agree with including the reasoning of "why" drag and drop should be included for this component. 

Just another note is that when drag and drop is eventually done, we have links from the drag and drop issue to this spec (and others) thanks to the cross linking so we can go through and implement it without digging around for what our intentions were.